### PR TITLE
12444 chapter31 veteran info updates

### PR DESCRIPTION
**Pull request by micahchiang**
## Description
This pull request updates the pattern used in the chapter 31 (VRE 28-1900) for toggling between authenticated veteran information and un-authenticated form fields. Unit tests and e2e tests will be written in a future pull request. 

This pull request exposes a recently non-exported function in `platform/forms-system/src/js/routing.js`, `getEligiblePages`, which allows the pageList passed into `SaveInProgressIntro` to be modified prior to entering the form. 

## Testing done
- local, form flow works properly authenticated and unauthenticated

## Screenshots
<details><summary>Unauthed review page</summary>

Screen Shot 2020-08-19 at 1 08 21 PM

</details>

<details><summary>Authed review page</summary>

Screen Shot 2020-08-19 at 1 59 13 PM

</details>

## Acceptance criteria
- [ ] form flows work as expected

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/platform/forms-system/src/js/routing.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ import { getActiveExpandedPages } from './helpers';
  * Returns the page list without conditional pages that have not satisfied
  * their dependencies and therefore should be skipped.
  */
-function getEligiblePages(pageList, data, pathname) {
+export function getEligiblePages(pageList, data, pathname) {
```
Comment (micahchiang) [inline src/platform/forms-system/src/js/routing.js:8]: I'm not aware of the reasoning behind why this function wasn't previously exposed, but there's a design pattern becoming more and more common in the frontend - where a form displays read-only veteran information for logged in users, and schemafields for logged out users - that benefits from having this function directly accessible. Calling this directly from the introduction page allows modification to the page list passed in to `<SaveInProgressIntro />`. This is helpful, especially when the first page in a form contains a `depends` attribute for toggling it on/off.

File: src/applications/vre/28-1900/config/form.js [hunk 2]
```diff
@@ -25,8 +29,38 @@ const formConfig = {
...
-        basicInformation: {
+        veteranStaticInformation: {
+          depends: () => hasSession(),
+          path: 'veteran-information-review',
+          title: 'Veteran Information Review',
+          hideHeaderRow: true,
+          schema: {
+            type: 'object',
+            properties: {
+              isLoggedIn: {
```
Comment (micahchiang) [inline src/applications/vre/28-1900/config/form.js:42]: I'll have a pr into `vets-api` here shortly that sends `isLoggedIn` back to the frontend as `true`, as part of the form prefill data. This flag allows conditional requirement of several fields in the logged out flow of the form.

**Review by cohnjesse — APPROVED** [review]
LGTM

File: src/platform/forms-system/src/js/routing.js (hunk 1 above)
Comment (cvalarida) [reply src/platform/forms-system/src/js/routing.js:8]: Ohhh, interesting. :thinking: 

I can't think of a reason this shouldn't be exported myself. Thanks for the explanation! :raised_hands:

**Review by cvalarida — APPROVED** [review]

**Pull request merged**